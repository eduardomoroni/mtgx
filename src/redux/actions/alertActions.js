import { SHOW_MESSAGE } from '../types'

export const showMessage = (message) => ({ type: SHOW_MESSAGE, message })
