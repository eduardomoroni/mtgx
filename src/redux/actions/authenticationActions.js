// @flow

import { SET_USER } from '../types'

export const setUser = (user: Object) => ({ type: SET_USER, user })
