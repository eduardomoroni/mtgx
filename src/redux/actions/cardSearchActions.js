import { SHOW_MESSAGE, CLEAR_MESSAGE } from '../types'

export const showMessage = (message) => ({ type: SHOW_MESSAGE, message })
export const clearMessage = () => ({ type: CLEAR_MESSAGE })
