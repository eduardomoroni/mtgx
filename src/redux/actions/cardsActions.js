// @flow

import { SET_CARDS } from '../types'

export const setCards = (cards: Object) => ({ type: SET_CARDS, cards })
