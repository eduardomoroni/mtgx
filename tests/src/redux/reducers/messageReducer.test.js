import { showMessage } from '../../../../src/redux/actions/alertActions'
import MessageReducer, { initialState } from '../../../../src/redux/reducers/messageReducer'

describe('Message reducer', () => {
  it('should have a snapshot', async () => {
    expect(initialState).toMatchSnapshot()
  })

  it('SHOW_MESSAGE action type should set message on state', async () => {
    const message = 'i love you, yeah you!'
    const action = showMessage(message)
    const state = MessageReducer(initialState, action)

    expect(state).toEqual(message)
  })
})
