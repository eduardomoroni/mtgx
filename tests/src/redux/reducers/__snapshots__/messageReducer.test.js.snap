// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message reducer should have a snapshot 1`] = `""`;
